header/interface.go

```diff
@@ -116,3 +115,11 @@ type Getter interface {
 	// GetRangeByHeight returns the given range [from:to) of ExtendedHeaders.
 	GetRangeByHeight(ctx context.Context, from, to uint64) ([]*ExtendedHeader, error)
 }
+
+// Head contains the behavior necessary for a component to retrieve
+// the chain head. Note that "chain head" is subjective to the component
+// reporting it.
+type Head interface {
```

Comment:
super unimportant nit, especially since I'm lacking context, but could there be a better name for the interface itself? For example, could something along the lines of `HeadRetriever` be more clear? But if we do this pattern in other places as well I would stick with it for consistency

Reply:
We have an interface called `Getter`

We could call it `HeadGetter` but I like the simplicity of it. I'm not too biased tho.

Reply:
yeah after thinking, the simplicity is nice, agreed